pkg/limayaml/validate.go

```diff
@@ -12,12 +12,13 @@ import (
 
 	"github.com/docker/go-units"
 	"github.com/lima-vm/lima/pkg/localpathutil"
+	"github.com/lima-vm/lima/pkg/networks"
 	"github.com/lima-vm/lima/pkg/osutil"
-	"github.com/lima-vm/lima/pkg/qemu/qemuconst"
+	"github.com/lima-vm/lima/pkg/qemu/const"
 	"github.com/sirupsen/logrus"
 )
 
-func Validate(y LimaYAML) error {
+func Validate(y LimaYAML, warn bool) error {
```

Comment:
Maybe we should make this like `Validate(y LimaYAML)(warns []error, err error)

Reply:
> Maybe we should make this like `Validate(y LimaYAML)(warns []error, err error)`

We could, but I'm not keen on it: there are over 50 `return` statements that would all need to deal with it:

```
$ grep return pkg/limayaml/validate.go | wc -l
      54
```

Just for the benefit of 3 possible warnings:

```
$ grep 'logrus\.' pkg/limayaml/validate.go
				logrus.Warnf("field `network.VDE` is deprecated; please use `networks` instead")
					logrus.WithError(err).Debugf("field `%s.vnl` %q failed stat", field, vdeSwitch)
					logrus.Warnf("field `%s.vnl` is unlikely to work for %s (unless libvdeplug4 has been ported to %s and is installed)",
```

And the warnings don't even come from `Validate` directly, but from `validateNetworks`.

I'll do it if you really think it makes a difference, but I think the code is clearer the way it currently is.